### PR TITLE
feat: expose unprotected metrics endpoint on internal port

### PR DESCRIPTION
**Pull request by alexmt**
Closes https://github.com/woodpecker-ci/woodpecker/issues/1477

PR exposes `/metrics` endpoint that does not require auth token on an internal port:

```
curl http://localhost:9001/metrics
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 2.575e-05
go_gc_duration_seconds{quantile="0.25"} 6.7542e-05
go_gc_duration_seconds{quantile="0.5"} 0.000106542
go_gc_duration_seconds{quantile="0.75"} 0.000162417
```

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

File: cmd/server/flags.go [hunk 1]
```diff
@@ -77,6 +77,12 @@ var flags = []cli.Flag{
 		Usage:   "grpc address",
 		Value:   ":9000",
 	},
+	&cli.StringFlag{
+		EnvVars: []string{"WOODPECKER_METRICS_SERVER_ADDR"},
+		Name:    "metrics-server-addr",
+		Usage:   "metrics server address",
+		Value:   ":9001",
```
Comment (anbraten) [inline cmd/server/flags.go:84]: Maybe we should disable it by default, so unaware users without firewalls don't leak any data?

File: docs/docs/30-administration/90-prometheus.md [hunk 2]
```diff
@@ -30,6 +30,12 @@ scrape_configs:
        - targets: ['woodpecker.domain.com']
 ```
 
+## Unauthenticated Access
+
+Alternatively, the unprotected `/metrics` endpoint is exposed on the internal port (9001 by default). Port is configurable via the `WOODPECKER_METRICS_SERVER_ADDR` environment variable.
+
+```diff
+
```
Comment (anbraten) [inline docs/docs/30-administration/90-prometheus.md:38]:
```suggestion
```

File: cmd/server/server.go [hunk 3]
```diff
@@ -258,6 +259,14 @@ func run(c *cli.Context) error {
 		})
 	}
 
+	if metricsServer := c.String("metrics-server-addr"); metricsServer != "" {
+		g.Go(func() error {
+			metricsRouter := gin.New()
+			metricsRouter.GET("/metrics", gin.WrapH(promhttp.Handler()))
+			return http.ListenAndServe(metricsServer, metricsRouter)
+		})
+	}
```
Comment (anbraten) [inline cmd/server/server.go:268]:
```suggestion
	if metricsServerAddr := c.String("metrics-server-addr"); metricsServerAddr != "" {
		g.Go(func() error {
			metricsRouter := gin.New()
			metricsRouter.GET("/metrics", gin.WrapH(promhttp.Handler()))
			return http.ListenAndServe(metricsServerAddr, metricsRouter)
		})
	}
```

File: docs/docs/30-administration/10-server-config.md [hunk 4]
```diff
@@ -160,6 +160,10 @@ Automatically generates an SSL certificate using Let's Encrypt, and configures t
 
 Configures the gRPC listener port.
 
+### `WOODPECKER_METRICS_SERVER_ADDR`
+> Default: `:9001`
+
+Configures the unprotected metrics listener port. Empty value disables the metrics server completely.
```
Comment (anbraten) [inline docs/docs/30-administration/10-server-config.md:166]:
```suggestion
Configures an unprotected metrics endpoint. An empty value disables the metrics endpoint completely.
```

File: cmd/server/flags.go (hunk 1 above)
Comment (alexmt) [reply cmd/server/flags.go]: Agree, unprotected metrics server might be a nasty surprise for someone. Changed default to empty and updated documentation accordingly.

**Comment by alexmt** [comment]
Thank you for the review, @anbraten ! Accepted your suggestions and disabled the unprotected metrics server by default. PTAL

**Review by anbraten — APPROVED** [review]

**Pull request merged**